src/c++/perf_analyzer/test_concurrency_manager.cc

```diff
@@ -249,6 +249,17 @@ class TestConcurrencyManager : public TestLoadManagerBase,
     CheckSequences(concurrency2);
   }
 
+  /// Test that tries to find deadlocks and livelocks
+  ///
+  void TestTimeouts()
+  {
+    WatchDog watchdog(1000);
```

Comment:
What exactly is the WatchDog doing here? How is it better than adding a sleep call directly here?

Reply:
The watchdog is a background thread that will raise an error after some amount of time (1000ms here). It is used to catch and report the case where a test hangs or is taking too long.